README.md

```diff
@@ -103,40 +112,66 @@ You will need to install the following packages:
 Additionally, there is an optional dependency to the MIT licensed cpptoml library, which is used to parse a RouDi config file for the mempool config.
 [cpptoml](https://github.com/skystrife/cpptoml)
 
-#### Build from sources
-
-iceoryx_utils and iceoryx_posh are deployed as independent cmake packages. Posh is using some functions from utils and is depending on it. You are able to build posh and utils and integrate in into existing cmake projects.
+### Build with CMake (all supported platforms)
 
-##### Build Script
-For the first start, we advise to use our build-test script for building everything.
+**NOTE:** Requires CMake version 3.14 or higher. If you only have CMake version 3.5 or higher available please use the
+build script.
 
+The `CMakeLists.txt` from `iceoryx_meta` can be used to easily develop iceoryx with an IDE.
+ 
+ 1. Clone the repository
+    ```
     git clone https://github.com/eclipse/iceoryx.git
-    ./tools/iceoryx_build_test.sh
-
-In default-mode, the script is not building the provided test. For a clean build just add "clean" as first argument to the script.
-
-If the script is not used, keep in mind to pass `-DTOML_CONFIG=on` to cmake if the optional RouDi config file feature shall be built.
-
-**NOTE:** Users who wish to build iceoryx with cmake can follow the below steps (Requires CMake 3.14 or higher).
-
-* Step into root folder of iceoryx.
-
-* Execute the below commands.
-
-```
-iceoryx$ cmake -Bbuild -Hiceoryx_meta // Customisation possible with cmake switches
-iceoryx$ cmake --build build
-```
-
-#### Build with tests
-
-To build iceoryx with tests, just add "test" as first argument to the script.
+    ```
+
+ 2. Generate the necessary build files
+    ```bash
+    cd iceoryx
+    cmake -Bbuild -Hiceoryx_meta -DTOML_CONFIG=ON
+    # when you have installed external dependencies like ncurses you have to add them
+    # to your prefix path
+    cmake -Bbuild -Hiceoryx_meta -DTOML_CONFIG=ON -DCMAKE_PREFIX_PATH=$(PWD)/build/dependencies/
+    ```
+    ```
+ 3. Compile the source code
+    ```
+    cmake --build build
+    ```
+
+With the following CMake switches you can add additional features:
+ 
+ |  switch  |  description |
+ |:---------|:-------------|
+ | ```dds``` | builds the iceoryx dds gateway which requires an installed CycloneDDS, see [https://github.com/eclipse-cyclonedds/cyclonedds](https://github.com/eclipse-cyclonedds/cyclonedds) |
+ | ```examples``` | builds all examples |
+ | ```introspection``` | the console introspection client which requires an installed ncurses library with terminfo support |
+ | ```test``` | enables module-, integration- and component-tests |
+ | ```TOML_CONFIG``` | activates config file support by using toml, if this is deactivated the central broker ```RouDi``` is not being build |
```

Comment:
This list will become outdated very soon as development continues. Can we not add the CMake option that lists all the available targets?

Reply:
This is the `iceoryx_meta` CMakeLists.txt documentation. Yes we maybe add new options there but then it is not a problem to keep this documentation up to date otherwise the user has to crawl through the Cmake file to see what configuration options are possible - and this is just horrible.